docs/book/src/reference/versions.md

```diff
@@ -1,175 +1,375 @@
 If someone is looking for opportunities to help with the project, this is definitely an area where additional hands
 and eyes will be more than welcome and greatly beneficial to the entire community.
 
-See the [following section](#kubernetes-version-support-as-a-function-of-cluster-topology) to understand how cluster topology affects version support.
+#### Maximum version skew between various Kubernetes components
 
-### Kubernetes Version Support As A Function Of Cluster Topology
+Standard [Kubernetes version Skew Policy](https://kubernetes.io/releases/version-skew-policy/)
+defines the maximum version skew supported between various Kubernetes components within a single cluster.
 
-The Core Provider, Kubeadm Bootstrap Provider, and Kubeadm Control Plane Provider run on the Management Cluster, and clusterctl talks to that cluster's API server.
+Notably, version skew between various Kubernetes components also defines constraints to be observed
+by Cluster API, Cluster API providers or Cluster API users when performing Kubernetes version upgrades.
 
-In some cases, the Management Cluster is separate from the Workload Clusters. The Kubernetes version of the Management and Workload Clusters are allowed to be different.
+In some cases, also Cluster API and/or Cluster API providers are defining additional version skew constraints. For instance:
+- If you are using kubeadm as a bootstrapper, you must abide to the [kubeadm skew policy](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/create-cluster-kubeadm/#version-skew-policy). 
+- If you are using image builder, all the Kubernetes components on a single machine are of the same version
+- If your Cluster has a defined topology, with `Cluster.spec.topology` set and referencing a ClusterClass, 
+  Cluster API enforces a single Kubernetes version for all the machines in the cluster.
 
-Management Clusters and Workload Clusters can be upgraded independently and in any order, however, if you are additionally moving from
-v1alpha3 (v0.3.x) or v1alpha4 (v0.4.x) to v1beta1 (v1.x) as part of the upgrade, prior to upgrading any workload cluster using Cluster API v1beta1, 
-the management cluster will need to be upgraded the at least the minimum supported Kubernetes version for your target CAPI version.
+## Supported versions matrix by provider or component
 
-These diagrams show the relationships between components in a Cluster API release (yellow), and other components (white).
+### Core provider (`cluster-api-controller`)
 
-#### Management And Workload Cluster Are the Same (Self-hosted)
+The following table defines the support matrix for the Cluster API core provider.
+See [Cluster API release support](#cluster-api-release-support) and [Kubernetes versions support](#kubernetes-versions-support).
 
-![Management/Workload Same Cluster](../images/management-workload-same-cluster.png)
+|                  | v1.8, _Maintenance Mode_ | v1.9              | v1.10             |
```

Comment:
Should this table be (also needs version changes)


```suggestion
|                  | v1.7, _Maintenance Mode_ | v1.8              | v1.9             |
```

to reflect the correct current versions? Or is this like this because its for main and supposed to be in the 1.10 book?